### PR TITLE
370 Pylint alerts corrections as part of intervention experiment

### PR DESCRIPTION
**Pull request by evidencebp**
Makes the interventions describe in [intervention issue](https://github.com/aajanki/yle-dl/issues/370).
The experiment is described [here](https://github.com/evidencebp/pylint-intervention/).

Each intervention was done in a dedicated commit with a message explaining it.

The PR is still *work in progress* and created now to present the changes and consult.

I fixed the bug in yledl\downloader.py that we discussed in the issue.
I also extracted functions in yledl\yledl.py in order to fix the too-many-statements alerts.

### CODE REVIEW

**Comment by aajanki** [comment]
Looking good! Can you move also other argument groups (`action_group`, `qual_group`, `dl_group`) from `arg_parser()` to new functions? Just to be consistent

File: yledl/yledl.py [hunk 1]
```diff
@@ -109,109 +109,12 @@ def open_config_file_and_remember_opened(filename: str, mode: str = 'r'):
...
-    )
-    io_group.add_argument(
-        '--no-create-dirs',
-        action='store_false',
-        dest='create_dirs',
-        help='Stop if an output directory does not exist',
-    )
+    _add_url_groups_arguments(url_group)
+    _add_io_group_arguments(io_group)
     action_group = io_group.add_mutually_exclusive_group()
```
Comment (aajanki) [inline yledl/yledl.py:118]: Include everything from here until the `--xattrs` in the `_add_io_group_arguments` function.
(action_group and resume_group are subgroups of io_group so all of them should go together.)

Comment (evidencebp) [reply yledl/yledl.py:118]: Before this PR I had a dilema - do the minimal change to remove the alert or handle the entire function.
I went with minimal impact but was not fully OK with that.
You solved my dilleman ;-)

I put each group in its own function (and resume_group was in the middle).
Is it OK?

Comment (aajanki) [reply yledl/yledl.py:118]: There seems to be circular dependencies that prevent code from running. For example, extractors.py imports stuff from areena_extractors.py, but areena_extractors.py imports something from extractors.py.

Maybe it's easier do this as two pull requests? Revert the extractor.py split here, then I can merge this without the split, and you can then do the splitting as a separate PR.

File: __init__.py
Comment (aajanki) [inline __init__.py:1]: I don't think this file is needed

File: yledl/yledl.py (hunk 1 above)
Comment (evidencebp) [reply yledl/yledl.py:118]: You are correct.
I reverted the splitting in this PR.

File: __init__.py
Comment (evidencebp) [reply __init__.py:1]: Correct again.
It was created and committed by mistake.
I deleted it.

Comment (aajanki) [reply __init__.py:1]: You deleted both the correct file and one extra file that *is* needed. ` __init__.py` at the repository root was not needed, but `yledl/__init__.py` is required (it specifies which functions are exported by the yledl module).

You can restore the correct state by reverting the commit [5603556](https://github.com/aajanki/yle-dl/pull/371/commits/5603556a4ae2e0b141fffd43506ff2dc3a175d41)

Comment (evidencebp) [reply __init__.py:1]: Sorry for that.
I reverted the commit.

**Review by aajanki — APPROVED** [review]
Looks good. Tests also pass.

I'll merge this now. You can do another pull request for further changes.

**Pull request merged**